src/passes/OptimizeInstructions.cpp

```diff
@@ -1170,12 +1169,8 @@ struct OptimizeInstructions
     optimizeMemoryAccess(curr->ptr, curr->offset, curr->memory);
     optimizeStoredValue(curr->value, curr->bytes);
     if (auto* unary = curr->value->dynCast<Unary>()) {
-      if (unary->op == WrapInt64) {
-        // instead of wrapping to 32, just store some of the bits in the i64
-        curr->valueType = Type::i64;
-        curr->value = unary->value;
```

Comment:
It looks like this optimization was applied even when we wouldn't otherwise be able to replace the `wrap` with its operand, so we should keep this and add new tests for it.

Reply:
Good catch! I read this code too quickly... restored and added testing that is not confused by the new optimization.